test/test_habitat_env.py

```diff
@@ -290,6 +291,74 @@ def test_rl_vectorized_envs(gpu2gpu):
                 ), "dones should be true after max_episode steps"
 
 
+@pytest.mark.parametrize("gpu2gpu", [False])
+def test_rl_vectorized_envs_batch_renderer(gpu2gpu):
+    import habitat_sim
+
+    if gpu2gpu and not habitat_sim.cuda_enabled:
+        pytest.skip("GPU-GPU requires CUDA")
+
+    configs, datasets = _load_test_data()
+    for config in configs:
+        with habitat.config.read_write(config):
+            config.habitat.simulator.enable_batch_renderer = True
+            config.habitat.simulator.habitat_sim_v0.enable_gfx_replay_save = (
+                True
+            )
+            config.habitat.simulator.create_renderer = False
+            config.habitat.simulator.habitat_sim_v0.gpu_gpu = gpu2gpu
+            agent_config = get_agent_config(config.habitat.simulator)
+            # Only keep the rgb_sensor
+            agent_config.sim_sensors = {
+                "rgb_sensor": agent_config.sim_sensors["rgb_sensor"]
+            }
+
+    num_envs = len(configs)
+    env_fn_args = tuple(zip(configs, datasets, range(num_envs)))
+    with habitat.VectorEnv(
+        make_env_fn=_make_dummy_env_func, env_fn_args=env_fn_args
+    ) as envs:
+        envs.initialize_batch_renderer(configs[0])
+        observations = envs.reset()
+        envs.post_step(observations)
+
+        for i in range(2 * configs[0].habitat.environment.max_episode_steps):
+            outputs = envs.step(
+                sample_non_stop_action_gym(envs.action_spaces[0], num_envs)
+            )
+            observations, rewards, dones, infos = [
+                list(x) for x in zip(*outputs)
+            ]
+
+            for env_obs in observations:
+                assert KEYFRAME_OBSERVATION_KEY in env_obs
+
+            observations = envs.post_step(observations)
+
+            for env_obs in observations:
+                assert KEYFRAME_OBSERVATION_KEY not in env_obs
+
+            assert len(observations) == num_envs
+            assert len(rewards) == num_envs
+            assert len(dones) == num_envs
+            assert len(infos) == num_envs
+
+            tiled_img = envs.render(mode="rgb_array")
+            new_height = int(np.ceil(np.sqrt(NUM_ENVS)))
+            new_width = int(np.ceil(float(NUM_ENVS) / new_height))
+            h, w, c = observations[0]["rgb"].shape
+            assert tiled_img.shape == (
```

Comment:
Anything easy we can do as far as asserting that something got rendered? Maybe we don't have handy image-comparison functions. But maybe we could compute and compare an average pixel value?

Reply:
I think that screenshot-based tests would be important to introduce here as we move forward.

Until we have that, comparing an aggregate could work.

We could also piggy-back other tests results to serve as ground truth. For example:
* Merge this test with `test_rl_vectorized_envs` by adding an additional `batch_renderer` parameter so that generated image lists can be compared after all runs.
* Or, add a parameter to this test for toggling the classing renderer to serve as a ground truth.

Reply:
You gave an emoji thumbs-up above but I don't see any new code to assert that something got rendered. Am I missing something? Or is this test not doable?

Reply:
I'm comparing pixel averages after the `reset()` call above.

This is so we have something to detect regressions until proper image comparison is introduced.